sdk/platform/README.md

```diff
@@ -0,0 +1,41 @@
+# Azure SDK Platform
+
+Azure SDK Platform contains two important components. The first component is related to HTTP and the second one is implementation for system specific features that Azure Core uses (like thread sleep).
+
+## HTTP Adapter
+
+Azure SDK has its own HTTP Request and Response structures. For this reason, an HTTP adapter is required to be implemented to use Azure SDK HTTP Request and Response as input for any HTTP Stack library (like libcurl or win32).
```

Comment:
Is what we are saying here true? Do we really have a transport policy and abstractions **because we have our own HTTP request and response structure**? That's what this paragraph reads like but I am not sure if that's the reason why we have this transport layer.

Reply:
yeah, that is the reason.  C99 does not offer a default HTTP request and HTTP stack to send/receive requests.
Our SDK brings an specific design for an HTTP Request and Response. And due to that, we need adapters for them to be used with any HTTP stack like libcurl.

We could use libcurl directly, and create our requests on a libcurl-ready way, but that would make us libcurl-only.  Having our on HTTP request makes be independent from any HTTP stack (but requires to build adapters for any of them)